### DISCUSSION TITLE
[Assignment 4] Should we really do two different implementations?

### DISCUSSION THREAD

**Discussion by Ahmadsm2005** [prosyslab-classroom/cs424-program-reasoning#440] [Q&A]
### Your Name

Ahmad

### Question

Hello! At first, I thought doing a bottom-up and a top-down solutions would improve my score due to, for example, the nature of the testcases of each problem would benefit each solution. However, I only used an optimized solution with heuristics of only one of them and it is getting a good score. Should I still implement two different methods even though one is doing well enough?

**Comment by KihongHeo** [comment]
숙제의 목표는 고득점이 아닙니다. 수업 내용을 잘 이해하기 위한 도구로서 의도가 있는 것이지요.

고작 점수에 본인을 최적화 하지마세요. 카이스트 수준에 어울리지 않습니다. 이제 대학교 4학년인데 그 보다 더 큰 그림을 스스로 그릴 줄 알아야겠지요. 시험 점수에만 최적화 되어있는 사람을 사회에서 필요로 할까요?

---
The goal of assignments is not to achieve high scores. They are intended as tools to help you better understand the course material.

Don’t optimize yourself solely for grades—that’s not fitting for the KAIST level. As a senior in university, you should be able to envision a bigger picture. Would society value someone who is optimized only for test scores?
